packages/react-slider/src/components/Slider/Slider.types.ts

```diff
@@ -1,33 +1,109 @@
 import * as React from 'react';
-import { ComponentPropsCompat } from '@fluentui/react-utilities';
+import { ComponentProps, ComponentState } from '@fluentui/react-utilities';
 
 /**
- * Slider Props
+ * Slider ref
  */
-export interface SliderProps extends ComponentPropsCompat, React.HTMLAttributes<HTMLElement> {
-  /*
-   * TODO Add props and slots here
-   * Any slot property should be listed in the sliderShorthandProps array below
-   * Any property that has a default value should be listed in SliderDefaultedProps as e.g. 'size' | 'icon'
+export interface SliderPublicRef {
+  /**
+   * Gets the current value of the Slider.
+   */
+  value: number | undefined;
+
+  /**
+   * Sets focus to the Slider's thumb.
    */
+  focus: () => void;
 }
 
 /**
  * Names of the shorthand properties in SliderProps
  */
-export type SliderShorthandProps = never; // TODO add shorthand property names
+export type SliderSlots = {
+  /**
+   * The **Slider's** base. It is used to visibly display the min and max selectable values.
+   */
+  rail: React.HTMLAttributes<HTMLElement>;
+
+  /**
+   * The wrapper around the Slider's track. It is primarily used to handle the positioning of the track.
+   */
+  trackWrapper: React.HTMLAttributes<HTMLElement>;
+
+  /**
+   * The bar showing the current selected area adjacent to the **Slider's** thumb.
+   */
+  track: React.HTMLAttributes<HTMLElement>;
+
+  /**
+   * The wrapper around the Slider's thumb. It is primarily used to handle the dragging animation from translateX.
+   */
+  thumbWrapper: React.HTMLAttributes<HTMLElement> & React.RefAttributes<HTMLDivElement>;
+
+  /**
+   * The draggable icon used to select a given value from the **Slider**.
+   */
+  thumb: React.HTMLAttributes<HTMLDivElement> & React.RefAttributes<HTMLDivElement>;
+
+  /**
+   * The area in which the **Slider's** rail allows for the thumb to be dragged.
+   */
+  activeRail: React.HTMLAttributes<HTMLElement> & React.RefAttributes<HTMLDivElement>;
+};
+
+export interface SliderCommon extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  /**
+   * The starting value for an `uncontrolled` **Slider**.
+   * Mutually exclusive with `value` prop.
+   */
+  defaultValue?: number;
```

Comment:
Most of these props overlap with native input props, which we're still trying to decide how to handle for consistency. What you have is basically [option 2 from the original RFC](https://github.com/microsoft/fluentui/pull/18804/files#diff-3598b99ac56bcedca9ac091588bddd4ad764a6cb57896f9b2d6ec90ea72e005bR101) which sounds good on the surface but nobody has been able to come up with a coherent definition for which props are "special." The weird thing for Slider is that there currently is no actual `input` element, but it seems like it probably still ought to follow the same pattern if possible.